lib/improver/nbhood/nbhood.py

```diff
@@ -391,3 +378,165 @@ def process(self, cube):
+        """
+        Create a neighbourhood processing plugin that generates percentiles
+        from a neighbourhood of points.
+
+        Parameters
+        ----------
+        neighbourhood_method : str
+            Name of the neighbourhood method to use. Options: 'circular',
+            'square'.
+        radii : float or List (if defining lead times)
+            The radii in metres of the neighbourhood to apply.
+            Rounded up to convert into integer number of grid
+            points east and north, based on the characteristic spacing
+            at the zero indices of the cube projection-x and y coords.
+        lead_times : None or List (optional)
+            List of lead times or forecast periods, at which the radii
+            within 'radii' are defined. The lead times are expected
+            in hours.
+        ens_factor : float (optional)
+            The factor with which to adjust the neighbourhood size
+            for more than one ensemble member.
+            If ens_factor = 1.0 this essentially conserves ensemble
+            members if every grid square is considered to be the
+            equivalent of an ensemble member.
+            Optional, defaults to 1.0
+        percentiles : list (optional)
+            Percentile values at which to calculate; if not provided uses
+            DEFAULT_PERCENTILES.
+        """
+        super(GeneratePercentilesFromANeighbourhood, self).__init__(
+            neighbourhood_method, radii, lead_times=lead_times,
+            ens_factor=ens_factor)
+
+        self.percentiles = percentiles
+
+        methods = {
+            "circular": GeneratePercentilesFromACircularNeighbourhood}
+        try:
+            method = methods[neighbourhood_method]
+            self.neighbourhood_method = method(percentiles=self.percentiles)
+        except KeyError:
+            msg = ("The neighbourhood_method requested: {} is not a "
+                   "supported method. Please choose from: {}".format(
+                       neighbourhood_method, methods.keys()))
+            raise KeyError(msg)
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        result = ('<{}; NeighbourhoodProcessing: percentiles: {}>')
+        return result.format(
+            super(GeneratePercentilesFromANeighbourhood, self).__repr__(),
+            self.percentiles)
+
+    def process(self, cube):
+        """
+        Apply percentile neighbourhood processing. This generates a set of
+        percentiles by using a neighbourhood of points around a central point.
+
+        Parameters
+        ----------
+        cube : Iris.cube.Cube
+            Cube to apply a neighbourhood processing method to, in order to
+            calculate percentile values from a neighbourhood.
+
+        Returns
+        -------
+        Iris.cube.Cube
+            Cube after calculating percentile values from a neighbourhood.
+        """
+        return self.run(cube)
```

Comment:
I can't see a reason for this. The same `process` method exists in `NeighbourhoodProcessing`. The same result would be achieved by renaming `BaseNeighbourhoodProcessing.run` as the `process` method and removing this method.

Reply:
Good point. I've removed this method.